src/shared/fdset.c

```diff
@@ -23,27 +24,29 @@ FDSet *fdset_new(void) {
         return MAKE_FDSET(set_new(NULL));
 }
 
-int fdset_new_array(FDSet **ret, const int *fds, size_t n_fds) {
-        size_t i;
-        FDSet *s;
+static inline void fdset_shallow_freep(FDSet **s) {
```

Comment:
Shouldn't we rename `fdset_free()` to `fdset_free_free()` and name this one `fdset_free()`, to mimick what we do with hashmaps and such?

Reply:
hmm, this is an internal helper, and only called in exceptional cases. And I think this is unlikely to change, given it really just exists to make our exported functions somewhat "atomic": either they succeed or they fail, but they won't take possesion of parameters on failure. Outside of the internal implementaiton we won't need that however.

I think "fdset_free()" as the primary exported function should use the generic name, and the internal, exceptional case get the special name.